core/types/receipt.go

```diff
@@ -295,26 +295,17 @@ type ReceiptForStorage Receipt
 func (r *ReceiptForStorage) EncodeRLP(_w io.Writer) error {
 	w := rlp.NewEncoderBuffer(_w)
 	outerList := w.List()
-	if r.Type == ArbitrumLegacyTxType {
-		w.WriteBytes(receiptRootArbitrumLegacy)
-		w.WriteUint64(r.CumulativeGasUsed)
-		w.WriteUint64(r.GasUsed)
-		w.WriteUint64(r.GasUsedForL1)
-		w.WriteUint64(r.Status)
-		rlp.Encode(w, r.ContractAddress)
```

Comment:
I think we need to maintain the old encoding here or else gas used for old legacy receipts are wrong. Basically in Arbitrum Classic the cumulative gas used didn't match up with the gas used nicely IIRC. We should try querying for old receipt gas used values on a migrated databases and make sure they match (particularly for classic blocks with multiple txs)